### PR TITLE
Don't write LIBARCHIVE.creationtime attribute in tar.gz archive entries

### PR DESCRIPTION
**Pull request by HannesWell**
Fixes https://github.com/eclipse-tycho/tycho/issues/2586

### CODE REVIEW

File: tycho-p2-director-plugin/src/main/java/org/eclipse/tycho/plugins/tar/TarGzArchiver.java [hunk 1]
```diff
@@ -129,6 +129,7 @@ private TarArchiveEntry createTarEntry(File tarRootDir, File source) throws IOEx
             tarEntry.setMode(FilePermissionHelper.toOctalFileMode(attrs.permissions()));
         }
         tarEntry.setModTime(source.lastModified());
+        tarEntry.setCreationTime(null); // tar cannot handle 'LIBARCHIVE.creationtime' attributes
```
Comment (laeubi) [inline tycho-p2-director-plugin/src/main/java/org/eclipse/tycho/plugins/tar/TarGzArchiver.java:132]: This is wrong, e.g. debian supports it: https://manpages.debian.org/unstable/libarchive-dev/tar.5.en.html

**Review by laeubi — CHANGES_REQUESTED** [review]
There should be a config option to switch this behavior on/off

- for Tycho 4.x it must be on by default
- for Tycho 5.x we can decide what should be the default

**Comment by HannesWell** [comment]
> There should be a config option to switch this behavior on/off
> 
>     * for Tycho 4.x it must be on by default
> 
>     * for Tycho 5.x we can decide what should be the default

If that's necessary get consensus I can add an option, but AFAIK an option was not added when this attribute was introduced, was it?

**Comment by laeubi** [comment]
> when this attribute was introduced

This attribute was never (intentionally) introduced, we updated the library behind that and those maintainer choose for some reasons to support this "non standard" extension. So from a compatibility point of view it only adds something new that can be ignored by a tool if desired.

No we have the other way round, that *removing* it might break use-cases that have relied on this, so I better like to preserve this. It would have been different if we would have never added it before but now it is there (even not added by intention).

**Comment by HannesWell** [comment]
> No we have the other way round, that _removing_ it might break use-cases that have relied on this, so I better like to preserve this. It would have been different if we would have never added it before but now it is there (even not added by intention) but disabling that by a config option is always possible of course.

Ok, then lets do it this way.
But I suggest to, for convenience to disable it by default for tycho-5 and keep the property by default for Tycho-4.
If one day tar can handle or ignore that attribute and the version that has the fix is widely adopted we can re-add it again.

File: tycho-p2-director-plugin/src/main/java/org/eclipse/tycho/plugins/tar/TarGzArchiver.java (hunk 1 above)
Comment (HannesWell) [reply tycho-p2-director-plugin/src/main/java/org/eclipse/tycho/plugins/tar/TarGzArchiver.java]: Understand. I refined it to `GNU tar`. I cannot provide the other details exhaustively.

**Comment by HannesWell** [comment]
@laeubi are you fine with this now?

**Comment by HannesWell** [comment]
Created the back-port with different default in
- https://github.com/eclipse-tycho/tycho/pull/4428

File: tycho-p2-director-plugin/src/main/java/org/eclipse/tycho/plugins/p2/director/ProductArchiverMojo.java [hunk 2]
```diff
@@ -125,6 +125,15 @@ public final class ProductArchiverMojo extends AbstractProductMojo {
...
     private boolean parallel;
 
+    /**
+     * Controls if for {@code .tar.gz} archives the creation-time is stored as
+     * {@code LIBARCHIVE.creationtime } attribute in each entry. Currently {@code GNU tar} does not
+     * support that attributes and emits warnings about the {@code unknown extended header keyword
+     * 'LIBARCHIVE.creationtime'} when extracting such archive.
+     */
+    @Parameter(defaultValue = "false")
+    private boolean storeTarGZCreationTime;
```
Comment (laeubi) [inline tycho-p2-director-plugin/src/main/java/org/eclipse/tycho/plugins/p2/director/ProductArchiverMojo.java:135]:
```suggestion
    private boolean storeCreationTime;
```

should be enough.

**Review by laeubi — APPROVED** [review]
Looks good!

**Pull request merged**